### PR TITLE
Form error

### PR DESCRIPTION
**Pull request by mhewson**
## Description
Adding Role tag="status to the generic error message for input and props for the slot that allow for status or alert- Modal also updated to allow role being changed from default "dialog" to alertdialog  

### A11y:
- [x] Meets WCAG AA standards

### CODE REVIEW

File: src/components/input/CdrInput.jsx [hunk 1]
```diff
@@ -230,7 +235,7 @@ export default {
             </span>
         )}
         {this.error && (
-          <cdr-form-error error={this.error}>
+          <cdr-form-error role={this.errorRole} error={this.error}>
```
Comment (cowills) [inline src/components/input/CdrInput.jsx:241]: does this properly let you override the role in formError? can't recall how native attrs work if it's hardcoded in the child but overridden in the parent

Comment (mhewson) [reply src/components/input/CdrInput.jsx:241]: It did in the examples I added on input, I'm not sure if I caught all the use-cases though?

**Comment by cowills** [comment]
select and formGroup also use formError so would need to be added in those places too presumably?

File: src/components/input/CdrInput.jsx (hunk 1 above)
Comment (cowills) [reply src/components/input/CdrInput.jsx:241]: yeah i think thats how it works i just cant test it at the moment 😆  as long as you can pass `errorRole="foobar"` to an input and see foobar in the formError role then :+1:

**Comment by mhewson** [comment]
Ah good call - I can take a look - the errors for now are defaulting to status for them - which they should be almost 99% of the time

**Review by cowills — APPROVED** [review]

**Pull request merged**